router/src/infer.rs

```diff
@@ -123,24 +123,11 @@ impl Infer {
                 err
             })?;
 
-        let mut adapter_id = request.parameters.adapter_id.clone();
-        if adapter_id.is_none() || adapter_id.as_ref().unwrap().is_empty() {
-            adapter_id = Some(BASE_MODEL_ADAPTER_ID.to_string());
-        }
-        let mut adapter_source = request.parameters.adapter_source.clone();
-        if adapter_source.is_none() {
-            adapter_source = Some(DEFAULT_ADAPTER_SOURCE.to_string());
-        }
-
-        let adapter_parameters =
-            request
-                .parameters
-                .adapter_parameters
-                .clone()
-                .unwrap_or(AdapterParameters {
-                    adapter_ids: vec![adapter_id.clone().unwrap()],
-                    ..Default::default()
-                });
+        let (adapter_id, adapter_source, adapter_parameters) = extract_adapter_params(
```

Comment:
I think we may want to just discard adapter_id at this point, as it should b contained in the adapter parameters.

So I would imagine this function just returning `adapter_source` and `adapter_parameters`.